src/cosalib/cmdlib.py

```diff
@@ -57,11 +55,27 @@ def run_verbose(args, **kwargs):
         kwargs['stdout'] = subprocess.PIPE
         kwargs['stderr'] = subprocess.PIPE
 
-    try:
-        process = subprocess.run(args, **kwargs)
-    except subprocess.CalledProcessError:
-        fatal("Error running command " + args[0])
-    return process
+    print("+ {}".format(subprocess.list2cmdline(args)))
+    with tempfile.NamedTemporaryFile(prefix="cmdlib-cmd") as t:
+        cosa_path = os.path.dirname(os.path.dirname(__file__))
+
+        # when cmdlib_sh=True, execute the command in a wrapper
+        # that sources the cmdlib.sh functions to allow using
+        # cmdlib.sh features like "runvm".
+        if kwargs.pop('cmdlib_sh', False):
```

Comment:
This seems fairly magical, in extending the subprocess API with a custom keyword.

Given how few things need this, maybe better done outside-in as:
```
def run_cmdlib(...):
   with tempfile:
      run_verbose(tmpshell)
```

?

Reply:
Agreed, esp. given that the cmdlib hack should really be a temporary thing until we move it all to Python.